computervision/src/main/java/ComputerVisionInAction.kt

```diff
@@ -16,16 +16,22 @@ internal class ComputerVisionInAction(private val cvPortal: SensorPortal) : Comp
         cv = cvPortal.connect()
     }
 
+
+
     private fun cvTokenize(rawData: String): Vision? {
         try {
             val raw = JSONObject(rawData)
 
+
             return when (raw.getInt("type")) {
                 OCR -> Vision.OCR(raw.getString("value"))
                 SCENES -> Vision.Scenery(raw.getString("value"))
                 PRETTY_FACES -> Vision.Facial(raw.getString("value"))
                 EMOTIONS -> Vision.Emotion(raw.getString("value"))
-                OBJECTS -> Vision.ObjectDetection(listOf<String>(raw.getString("value"))) //[[......]])
+                OBJECTS ->Vision.ObjectDetection(((0 until raw.getJSONArray("values").length()).map {
+                    DetectedObject(raw.getJSONArray("values").getJSONObject(it).getString("label"),
+                        raw.getJSONArray("values").getJSONObject(it).getString("pos") )}))
+
```

Comment:
Brilliant! You used `map` on your own. Even better, you used it on a `Range`, which means you did your homework on `JSONArray`.
Now you’re officially on a new level—You can search for new “code” on your own without any guidance. This actually is a huge step up!

That being said, tiny mistakes will no longer be tolerated; Don’t you think `getJSONArray` is called quite too much in this code? This needs to change.

Reply:
😮 🙇‍♀  🙇‍♀